test/ForwardMode/ConstFunctions.C

```diff
@@ -0,0 +1,33 @@
+// RUN: %cladclang %s -I%S/../../include -oConstFunctions.out 2>&1 | FileCheck %s
+// RUN: ./ConstFunctions.out | FileCheck -check-prefix=CHECK-EXEC %s
+// CHECK-NOT: {{.*error|warning|note:.*}}
+
+#include "clad/Differentiator/Differentiator.h"
+
+#include "../TestUtils.h"
+
+#define TEST(name,i,j,k) \
+  auto d_##name = clad::differentiate(name,"x");\
+  printf("%.2f\n", d_##name.execute(i,j,k));\
+  printf("%.2f\n", d_##name.execute(i,j,k));\
+  printf("\n");\
+
+constexpr double fn1(double x, double y, double z){
+	return x*y*z+x+y+z;
+}
+
+//CHECK:    constexpr double fn1_darg0(double x, double y, double z) {
+//CHECK-NEXT:    double _d_x = 1;
+//CHECK-NEXT:    double _d_y = 0;
+//CHECK-NEXT:    double _d_z = 0;
+//CHECK-NEXT:    double _t0 = x * y;
+//CHECK-NEXT:    return (_d_x * y + x * _d_y) * z + _t0 * _d_z + _d_x + _d_y + _d_z;
+//CHECK-NEXT:}
+
+// consteval double fn2(double x, double y, double z){
+// 	return x*y*z;
+// }
+
+int main(){
+    TEST(fn1,3,4,5); //CHECK-EXEC: 21.00
```

Comment:
You can remove the `TEST` macro, and use `INIT_DIFFERENTIATE` and `TEST_DIFFERENTIATE` functions instead. They are defined in `TestUtils.h` file. 

```suggestion
   INIT_DIFFERENTIATE(fn1, "x");
   TEST_DIFFERENTIATE(fn1, 3, 4, 5);
```

Reply:
will do so